tests/jq.test

```diff
@@ -1834,3 +1834,26 @@ null
 debug
 1
 1
+
+# try/catch catches more than it should #1859
+(try . catch .) | error
+"foo"
+%%ERROR: "foo"
+
+.[]|(try (if .=="hi" then . else error end) catch empty) | "\(.) there!"
+["hi","ho"]
+"hi there!"
+
+.[]|(try . catch (if .=="ho" then "BROKEN"|error else empty end)) | if .=="ho" then error else "\(.) there!" end
+["hi","ho"]
+"hi there!"
+%%ERROR: "ho"
```

Comment:
It's `%%FAIL IGNORE MSG<cr>input`, or `%%FAIL<cr>input<cr>exact error`, not `%%ERROR: line pattern`. Also note that you can only check `stderr` that must be one line long when you use `%%FAIL`.

Reply:
Yup, like I said, it was a straightforward cherry-pick.  I don't recall if this was different on the `dlopen` branch, and I've not had time today to look after cherry-picking.

Reply:
I've not looked at the `dlopen()` branch, but anyways this is fixed now.